src/connector/src/sink/file_sink/opendal_sink.rs

```diff
@@ -110,7 +110,13 @@ impl<S: OpendalSinkBackend> Sink for FileSink<S> {
                 "File sink only supports `PARQUET` encode at present."
             )));
         }
-        Ok(())
+
+        match self.op.list(&self.path).await {
```

Comment:
https://github.com/risingwavelabs/risingwave/issues/18159

maybe leave a comment to indiate that this is a validation step

or move it to a dedicated "validte" function so that later people will remember to consolidate it in a unified interface

Reply:
This function is called `validate` 🤣